scripts/scil_sh_to_aodf.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Script to estimate asymmetric ODFs (aODFs) from a spherical harmonics image.
+
+Two methods are available:
+    * Angle-aware bilateral filtering [1] is an extension of bilateral
+      filtering considering the angular distance between sphere directions
+      for filtering 5-dimensional spatio-angular images.
+    * Cosine filtering [2] is a simpler implementation using cosine distance
+      for assigning weights to neighbours.
+
+Angle-aware bilateral filtering can be performed on the GPU using pyopencl by
+specifying --use_gpu. Make sure you have pyopencl installed to use this option.
+Otherwise, the filtering runs entirely on the CPU.
+"""
+
+import argparse
+import logging
+import time
+import nibabel as nib
+import numpy as np
+
+from dipy.data import SPHERE_FILES
+from dipy.reconst.shm import sph_harm_ind_list
+from scilpy.reconst.utils import get_sh_order_and_fullness
+from scilpy.io.utils import (add_overwrite_arg, add_verbose_arg,
+                             assert_inputs_exist, add_sh_basis_args,
+                             assert_outputs_exist)
+from scilpy.denoise.asym_filtering import (cosine_filtering,
+                                           angle_aware_bilateral_filtering)
+
+
+EPILOG = """
+[1] Poirier et al, 2022, "Intuitive Angle-Aware Bilateral Filtering Revealing
+    Asymmetric Fiber ODF for Improved Tractography", ISMRM 2022 (abstract 3552)
+
+[2] Poirier et al, 2021, "Investigating the Occurrence of Asymmetric Patterns
+    in White Matter Fiber Orientation Distribution Functions", ISMRM 2021
+    (abstract 0865)
+"""
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__, epilog=EPILOG,
+                                formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('out_sh',
+                   help='File name for averaged signal.')
+
+    p.add_argument('--out_sym', default=None,
+                   help='Name of optional symmetric output. [%(default)s]')
+
+    add_sh_basis_args(p)
+
+    p.add_argument('--sphere', default='repulsion724',
+                   choices=sorted(SPHERE_FILES.keys()),
+                   help='Sphere used for the SH to SF projection. '
+                        '[%(default)s]')
+
+    p.add_argument('--method', default='bilateral',
+                   choices=['bilateral', 'cosine'],
```

Comment:
Just to know. Is bilateral really the most common? Do we need a default here? Or should we make sure that user knows which method he's using by forcing him to choose?

Reply:
I would not say it is the most common (because neither one is really used), but it is the one I recommend. It's similar to the one described in my neuroimage paper and therefore it is better validated than the other implementation.